connector/src/main/java/org.apache.flink/connector/nebula/catalog/NebulaCatalog.java

```diff
@@ -0,0 +1,240 @@
+            CatalogException {
+        if (listDatabases().contains(databaseName.trim())) {
+            Map<String, String> props = new HashedMap();
+            try {
+                props.put("spaceId",
+                        String.valueOf(metaClient.getSpace(databaseName).getSpace_id()));
+            } catch (TException e) {
+                LOG.error("get spaceId error, ", e);
+            }
+            return new CatalogDatabaseImpl(props, databaseName);
+        } else {
+            throw new DatabaseNotExistException(getName(), databaseName);
+        }
+    }
+
+    /**
+     * objectName in tablePath mush start with VERTEX. or EDGE.
+     *
+     * @param tablePath A graphSpace name and label name.
+     * @return
+     */
+    @Override
+    public boolean tableExists(ObjectPath tablePath) throws CatalogException {
+        String graphSpace = tablePath.getDatabaseName();
+        String table = tablePath.getObjectName();
+        try {
+            return (listTables(graphSpace).contains(table));
+        } catch (DatabaseNotExistException e) {
+            throw new CatalogException("failed to call tableExists function, ", e);
+        }
+    }
+
+
+    /**
+     * show all tags and edges
+     *
+     * @param graphSpace nebula graph space
+     * @return List of Tag and Edge, tag starts with VERTEX. and edge starts with EDGE. .
+     */
+    @Override
+    public List<String> listTables(String graphSpace) throws DatabaseNotExistException,
+            CatalogException {
+        if (!databaseExists(graphSpace)) {
+            throw new DatabaseNotExistException(getName(), graphSpace);
+        }
+
+        try {
+            metaClient.connect();
+        } catch (TException e) {
+            LOG.error(String.format("failed to connect meta service vis %s ", address), e);
+            throw new CatalogException("nebula meta service connect failed.", e);
+        }
+        List<String> tables = new ArrayList<>();
+        try {
+            for (TagItem tag : metaClient.getTags(graphSpace)) {
+                tables.add("VERTEX" + NebulaConstant.POINT + tag.tag_name);
+            }
+            for (EdgeItem edge : metaClient.getEdges(graphSpace)) {
+                tables.add("EDGE" + NebulaConstant.POINT + edge.edge_name);
+            }
+        } catch (TException | ExecuteFailedException e) {
+            LOG.error("get tags or edges error,", e);
+        }
+        return tables;
+    }
+
+    @Override
+    public CatalogBaseTable getTable(ObjectPath tablePath) throws TableNotExistException,
+            CatalogException {
+        if (!tableExists(tablePath)) {
+            throw new TableNotExistException(getName(), tablePath);
+        }
+
+        String graphSpace = tablePath.getDatabaseName();
+        String[] typeAndLabel = tablePath.getObjectName().split(NebulaConstant.SPLIT_POINT);
+        String type = typeAndLabel[0];
```

Comment:
any chance OutOfBound here?

Reply:
no chance OutOfBound, the check will stop it.